artistools/inputmodel/maptogrid.py

```diff
@@ -429,12 +440,32 @@ def addargs(parser: argparse.ArgumentParser) -> None:
     parser.add_argument(
         "-ncoordgrid", type=int, default=50, help="Number of grid positions per axis (numcells = ncoordgrid^3)"
     )
+    parser.add_argument(
+        "-dtextra_seconds",
+        type=float,
+        default=0.5,
+        help="Time in seconds to propagate SPH particles ballistically after end of SPH simulation."
+        " 0 for no extrapolation",
+    )
+    parser.add_argument(
+        "-setgrid_fractionrmax",
+        type=float,
+        default=0.5,
+        help="Setup grid to have max equal to fraction of particle rmax. Default is 50% of rmax.",
+    )
     parser.add_argument(
         "-downsamplefactor",
         type=int,
         default=1,
         help="Randomly sample particles, reducing the number by this factor (e.g. 2 will ignore half of the particles)",
     )
+    parser.add_argument(
+        "-modifysmoothinglength",
```

Comment:
Unless a bunch of options are planned ("option4" is not a descriptive name), I would replace this with a boolean flag like --limit_outer_smoothinglength

Reply:
I'm planning a bunch of options. This needs played around with for different models

Reply:
Ideally I'd like to print to the log file what option was used here to keep a record of it

Reply:
Then you could have choices like -smoothinglength normal or -smoothinglength limitouter.

Reply:
The problem with that is there are lots of ways to increase the outer smoothing length. There's no right way to do this. It's really about keeping a record of the different ways to try